genrl/agents/deep/a2c/a2c.py

```diff
@@ -105,13 +105,11 @@ def select_action(
             value (:obj:`torch.Tensor`): Value of given state
             log_prob (:obj:`torch.Tensor`): Log probability of selected action
         """
-        state = torch.as_tensor(state).float().to(self.device)
-
         # create distribution based on actor output
         action, dist = self.ac.get_action(state, deterministic=deterministic)
         value = self.ac.get_value(state)
 
-        return action.detach().cpu().numpy(), value, dist.log_prob(action).cpu()
+        return action.detach(), value, dist.log_prob(action).cpu()
```

Comment:
Any reason to put it on the cpu? We could avoid these cpu calls here and just shift the tensor to the cpu before stepping into the env

Reply:
`select_action` is only used when pushing actions into the environment. I think it belongs more here than in the environment call as that would make the code a little confusing. It's fine either way this seemed more intuitive to me. Your call.

Reply:
> `select_action` is only used when pushing actions into the environment.

Is this correct? For e.g. if we have our replay/rollout buffer on the GPU (which is now possible), it could increase the performance by A LOT. Because we skip a whole lot of communication.